src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/EditSavesModal.tsx

```diff
@@ -0,0 +1,153 @@
+import * as Yup from "yup"
+import { pick } from "lodash"
+import { useUpdateCollection } from "./Mutations/useUpdateCollection"
+import createLogger from "Utils/logger"
+
+interface EditSavesModalProps {
+  collection: SavesArtworks_collection$data
+  setIsEditModalOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+interface FormikValues {
+  name: string
+}
+
+const MAX_NAME_LENGTH = 40
+
+const logger = createLogger("EditSavesModal")
+
+export const EditSavesModal: React.FC<EditSavesModalProps> = ({
+  collection,
+  setIsEditModalOpen,
+}) => {
+  const { t } = useTranslation()
+
+  const initialValues = pick(collection, "name")
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+      .required(t("collectorSaves.editListModal.fields.name.required")) // TODO: confirm copy
+      .max(MAX_NAME_LENGTH),
+  })
+
+  const { submitMutation } = useUpdateCollection()
+
+  const { sendToast } = useToasts()
+
+  const handleSubmit = async (
+    formikValues: FormikValues,
+    formikHelpers: FormikHelpers<FormikValues>
+  ) => {
+    try {
+      await submitMutation({
+        variables: {
+          input: {
+            id: collection.internalID,
+            name: formikValues.name,
+          },
+        },
+        rejectIf: res => {
+          const result = res.updateCollection?.responseOrError
+
+          return result?.__typename === "UpdateCollectionFailure"
+            ? result.mutationError
+            : false
+        },
+      })
+
+      sendToast({
+        variant: "success",
+        message: t("collectorSaves.editListModal.success"),
+      })
+    } catch (error) {
+      logger.error(error)
+      sendToast({
+        variant: "error",
+        message: error.message ?? t("common.errors.somethingWentWrong"),
+      })
+    } finally {
+      formikHelpers.setSubmitting(false)
+      setIsEditModalOpen(false)
+    }
+  }
+
+  return (
+    <Formik<FormikValues>
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({
+        values,
+        errors,
+        touched,
+        handleBlur,
+        handleChange,
+        isSubmitting,
+      }) => {
+        return (
+          <ModalDialog
+            title={t("collectorSaves.editListModal.title")}
+            onClose={() => {
+              setIsEditModalOpen(false)
+            }}
+            width={["100%", 700]}
+          >
+            <Form /* TODO: confirm implicit form submission */>
+              <LabeledInput
+                name="name"
+                value={values.name}
+                title={t("collectorSaves.editListModal.fields.name.label")}
+                label={<EditIcon />}
+                error={touched.name && errors.name}
+                maxLength={MAX_NAME_LENGTH}
+                required // TODO: confirm
+                onChange={handleChange}
+                onBlur={handleBlur}
+              />
```

Comment:
I think we should trim spaces in the collection name. otherwise, this situation may happen. the same should be done for create list flow (cc: @nickskalkin)

https://user-images.githubusercontent.com/3513494/222706862-bce9d725-142c-461d-88be-36aaf8b38da6.mp4

Reply:
Added in 61cb84bf73